programs/zstdcli.c

```diff
@@ -544,6 +544,14 @@ static int init_cLevel(void) {
     return ZSTDCLI_CLEVEL_DEFAULT;
 }
 
+#define ZSTD_CLI_STATIC_ASSERT(c) (void)sizeof(char[(c) ? 1 : -1])
+
+#define ZSTD_NB_STRATEGIES ZSTD_STRATEGY_MAX
```

Comment:
I'm surprised this works, because there is no place where `zstdcli.c` depends explicitly on experimental functions within `zstd.h`.

It appears this capability is inherited transitively from `benchzstd.h`.

But this is not clean: `benchzstd.h` could change its dependencies anytime in the future and no longer depend on experimental declarations. Therefore, if `zstdcli.c` depends on experimental functions, it should be explicitly mentioned in its own list of imports.

Add the macro `ZSTD_STATIC_LINKING_ONLY` in front of `#include "zstd.h"`. Explain why it's needed (`ZSTD_STRATEGY_MAX`).